Add Guidelines to become an Airflow Committer (#13236)
mistercrunch/Airflow: CONTRIBUTING.rst

    the mentor must try to remain impartial and cannot be the Proposer.

Inactive Committers
-------------------
If you know you are not going to be able to contribute for a long time

> (for instance due to a change of job or circumstances) you should inform the PMC and we will mark you

Reviewer: ```suggestion
(for instance, due to a change of job or circumstances), you should inform the PMC and we will mark you
```

as "inactive". Inactive committers will be removed from the "roster" on ASF and will no longer have the power
of being a Committer (especially write access to the repos). As merit earned never expires, once you
become active again you can simply email the PMC and ask to be reinstated.

The PMC also can mark committers as inactive after they have not been involved in the community for
more than 12 months.

> Guidelines to become an Airflow Committer

Reviewer: Is is related to "Committer Responsibilities" section?
https://github.com/apache/airflow/blob/6f78c4b909de875809825f18c1873e724bd20ce6/CONTRIBUTING.rst#committer-responsibilities

Reviewer 2: I guess we might link to it. 

This is rather different level but they describe what "active" committer should be responsible for.

------------------------------------------

> Committers are community members that have write access to the project’s

Reviewer: ```suggestion
Committers are community members who have write access to the project’s
```

repositories, i.e., they can modify the code, documentation, and website by themselves and also
accept other contributions. There is no strict protocol for becoming a committer. Candidates for new
committers are typically people that are active contributors and community members.

Some people might be active in several of those areas and while they might have not enough 'achievements' in any
single one of those, their combined contributions in several areas all count.

> As a community we appreciate contributions to the Airflow codebase, but we also place equal value

Reviewer: ```suggestion
As a community, we appreciate contributions to the Airflow codebase, but we also place equal value
```

on those who help Airflow by improving the community in some way. It is entirely possible to become
a committer (and eventually a PMC member) without ever having to change a single line of code.

Prerequisites
^^^^^^^^^^^^^^

General prerequisites that we look for in all candidates:

> 1.  Consistent contribution over last few months

Reviewer 2: :heart:

> 2.  Visibility on discussions on the dev mailing list and/or the Slack channel

Reviewer 3: Just for complement sake, should we add something like:
```suggestion
2.  Visibility on discussions on the dev mailing list, Slack channels or Github issues/discussions
```

Btw `and/or == or` 😄

Author: 😄  indeed

Author: Updated in e729fc1

> 3.  Helped to improve community of Airflow to be stronger

Reviewer: ```suggestion
3.  Helped to improve a community of Airflow to be stronger
```

4.  Understands contributor/committer guidelines : Contributors' Guide

Code contribution
^^^^^^^^^^^^^^^^^^

> 1.  Makes high quality commits (especially commit messages), and assess the impact of the changes including

Reviewer: ```suggestion
1.  Makes high-quality commits (especially commit messages), and assess the impact of the changes, including
```

    upgrade paths or deprecation policies
2.  Testing Release Candidates to help the release cycle

> 3.  Proposed and led to completion Airflow Improvement Proposal(s)

Reviewer 3: ```suggestion
3.  Proposed and led to completion of Airflow Improvement Proposal(s)
```
But I'm not a native speaker

Author: I think this one is fine

4.  Demonstrates an understanding of Airflow Core, API, Docker image, Helm chart, dev tools or certain

>     Providers or has displayed holistic understanding of a particular part and made contributions towards a more

Reviewer: ```suggestion
    Providers or has displayed a holistic understanding of a particular part and made contributions towards a more
```

>     Providers or has displayed a holistic understanding of a particular part and made contributions towards a more

Reviewer 3: As we are enumerating the areas, maybe it would be feasible to put them in bullet list for better readability?

Author: Updated in https://github.com/apache/airflow/pull/13236/commits/e729fc1bc86cf3f5369e41f1162c95ab1eb6ee07

    strategic goal

> 5.  Has made significant improvement or added an integration with services/technologies important to the Airflow

Reviewer: ```suggestion
5.  Has made a significant improvement or added an integration with services/technologies important to the Airflow
```

    Ecosystem

Community contributions
^^^^^^^^^^^^^^^^^^^^^^^^

> 1.  Was instrumental in triaging issues.

Reviewer 4: ```suggestion
1.  Was instrumental in triaging issues
```

To be consistent with other lines.

2.  Improved documentation of Airflow in significant way
3.  Lead change and improvements introduction in the “community” processes and tools

> 4.  Actively spreads the word about Airflow, example organising Airflow summit, workshops for

Reviewer 4: ```suggestion
4.  Actively spreads the word about Airflow, for example organising Airflow summit, workshops for
```

    community members, giving and recording talks, writing blogs

> 5.  Reporting bugs with details reproduction steps

Reviewer: ```suggestion
5.  Reporting bugs with detailed reproduction steps
```

more than 12 months.

Guidelines to become an Airflow Committer
------------------------------------------

Committers are community members who have write access to the project’s
repositories, i.e., they can modify the code, documentation, and website by themselves and also
accept other contributions. There is no strict protocol for becoming a committer. Candidates for new
committers are typically people that are active contributors and community members.

Some people might be active in several of those areas and while they might have not enough 'achievements' in any
single one of those, their combined contributions in several areas all count.

As a community, we appreciate contributions to the Airflow codebase, but we also place equal value
on those who help Airflow by improving the community in some way. It is entirely possible to become
a committer (and eventually a PMC member) without ever having to change a single line of code.

Prerequisites
^^^^^^^^^^^^^^

General prerequisites that we look for in all candidates:

1.  Consistent contribution over last few months
2.  Visibility on discussions on the dev mailing list, Slack channels or Github issues/discussions

> 3.  Helped to improve community of Airflow to be stronger

Reviewer 5: Maybe we can rephrase this a bit?
```suggestion
3.  Contributions to community health and project's sustainability for the long-term
```
